### DISCUSSION TITLE
Change to APP_URL in .env doesn't change the url in Ziggy

### DISCUSSION THREAD

**Discussion by cosnofski** [tighten/ziggy#644] [Q&A]
Scratching my head over this. But if I make a change to APP_URL in my .env file, the generated javascript in app.blade.php doesn't change instead it remains as the old APP_URL. What am I missing?

I've tried clearing application cache, configuration cache and clearing views but nothing helps

```
php artisan cache:clear
php artisan config:clear
php artisan view:clear
```

The generated javascript in app.blade.php stays the same 

When I run

```
echo config('app.url'); // prints new APP_URL
echo url('/'); // prints new url
Log::debug(print_r($_ENV, true)); // Prints new APP_URL to log
```

**Comment by bakerkretzmar** [comment]
Ziggy gets its URL from `url('/')`, not the `APP_URL` environment variable, so the Blade directive will always use the request URL unless you override it:

https://github.com/tighten/ziggy/blob/main/src/Ziggy.php#L27

**Comment by cosnofski** [comment]
Thanks for the response. Is there an easy way to override the url? If not, seems like I should create my own blade decorator that mimics `Tightenco\Ziggy\BladeRouteGenerator` and does something more or less like:

```
$ziggy = (new Ziggy($group, config('app.url')));
$output = config('ziggy.output.file', File::class);
$str = (string) new $output($ziggy);
```

Thanks again

**Reply by bakerkretzmar** [reply]
Not right in the Blade directive no, so you'd need to either do something like what you showed or override it in a script right after:

```blade
@routes
<script>
    Ziggy.url = 'you-url';
</script>
```
